### PR TITLE
Chore: Migrate to k3d in CI & trim redundant steps

### PR DESCRIPTION
**Pull request by chivalryq**
Signed-off-by: Qiaozp <qiaozhongpei.qzp@alibaba-inc.com>

Speedup：

1. Multicluster e2e 25min -> 10min
2. **e2e 44min -> 25min**
3. APIServer e2e 20min -> 10min
4. Rollout E2e test 35min -> 25min

### Description of your changes

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: charts/vela-core/templates/test/test-application.yaml [hunk 1]
```diff
@@ -13,12 +13,6 @@ spec:
       properties:
         image: {{ .Values.imageRegistry }}{{ .Values.test.app.repository }}:{{ .Values.test.app.tag }}
         port: 8000
-      traits:
-        - type: ingress
```
Comment (wonderflow) [inline charts/vela-core/templates/test/test-application.yaml:17]: change to gateway?

Comment (chivalryq) [reply charts/vela-core/templates/test/test-application.yaml:17]: would fail on 1.20

Comment (wonderflow) [reply charts/vela-core/templates/test/test-application.yaml:17]: 👌🏻

File: .github/workflows/apiserver-test.yaml [hunk 2]
```diff
@@ -136,13 +136,13 @@ jobs:
       - name: Setup K3d (Hub)
         uses: nolar/setup-k3d-k3s@v1.0.8
         with:
-          version: v1.18
+          version: v1.20
```
Comment (wonderflow) [inline .github/workflows/apiserver-test.yaml:139]: let's make the version as a global value in action.

The same below

Comment (chivalryq) [reply .github/workflows/apiserver-test.yaml:139]: Yes, this version is only for test now.

**Review by wonderflow — APPROVED** [review]

**Pull request merged**